sdk/python/sawtooth_sdk/processor/core.py

```diff
@@ -33,65 +44,174 @@
+                    family=n,
+                    version=v,
+                    encoding=e,
+                    namespaces=h.namespaces)
+                 for n, v, e in itertools.product(
+                    [h.family_name],
+                     h.family_versions,
+                     h.encodings)] for h in self._handlers])
 
-            request = TpProcessRequest()
-            request.ParseFromString(msg.content)
-            state = State(self._stream, request.context_id)
+    def _unregister_requests(self):
+        """Returns all of the TpUnRegisterRequests for handlers
+        :note: itertools.chain.from_iterable flattens the nested lists
+        :return (list): list of TpUnRegisterRequests
+        """
+        return itertools.chain.from_iterable(
+            [
+                [TpUnRegisterRequest(
+                    family=n,
+                    version=v,
+                    encoding=e,
+                    namespaces=h.namespaces)
+                 for n, v, e in itertools.product(
+                    [h.family_name],
+                    h.family_versions,
+                    h.encodings)] for h in self._handlers])
 
+    def _process(self, msg):
+        request = TpProcessRequest()
+        request.ParseFromString(msg.content)
+        state = State(self._stream, request.context_id)
+        header = TransactionHeader()
+        header.ParseFromString(request.header)
+        try:
+            if not self._stream.is_ready():
+                raise ValidatorConnectionError()
+            self._find_handler(header).apply(request, state)
+            self._stream.send_back(
+                message_type=Message.TP_PROCESS_RESPONSE,
+                correlation_id=msg.correlation_id,
+                content=TpProcessResponse(
+                    status=TpProcessResponse.OK
+                ).SerializeToString())
+        except InvalidTransaction as it:
+            LOGGER.warning("Invalid Transaction %s", it)
             try:
-                self._handlers[0].apply(request, state)
-                self._stream.send_back(
-                    message_type=Message.TP_PROCESS_RESPONSE,
-                    correlation_id=msg.correlation_id,
-                    content=TpProcessResponse(
-                        status=TpProcessResponse.OK
-                    ).SerializeToString())
-            except InvalidTransaction as it:
-                LOGGER.warning("Invalid Transaction %s", it)
                 self._stream.send_back(
                     message_type=Message.TP_PROCESS_RESPONSE,
                     correlation_id=msg.correlation_id,
                     content=TpProcessResponse(
                         status=TpProcessResponse.INVALID_TRANSACTION
                     ).SerializeToString())
-            except InternalError as ie:
-                LOGGER.warning("State Error! %s", ie)
+            except ValidatorConnectionError as vce:
+                LOGGER.warning("%s", vce)
```

Comment:
Can we provide more context here?  Tracking down where this was output would be challenging if we don't add any unique text.